### PR TITLE
Clean up and format code in preparation for 8.0 release and Magento Marketplace approval

### PR DESCRIPTION
**Pull request by jantzenw**

### CODE REVIEW

File: Model/Source/ProductList.php [hunk 1]
```diff
@@ -26,33 +19,36 @@ class ProductList
...
-            ),
-            array(
+                'label' => __('Category and Search Result Pages'),
+            ],
+            [
                 'value' => self::UPSELL,
-                'label' => __('Upsells on Product Pages')
-            ),
-            array(
+                'label' => __('Upsells on Product Pages'),
```
Comment (brobie) [inline Model/Source/ProductList.php]: In Magento, they use the name format of "Up-Sell Products" (dash)

File: Model/Source/ProductList.php [hunk 2]
```diff
@@ -26,33 +19,36 @@ class ProductList
...
+            [
                 'value' => self::CROSSSELL,
-                'label' => __('Cross Sells on Cart Page')
-            ),
-	        array(
-		        'value' => self::WIDGET,
-		        'label' => __('Product List Widget')
-	        ),
-        );
+                'label' => __('Cross Sells on Cart Page'),
```
Comment (brobie) [inline Model/Source/ProductList.php]: In Magento, they use the name format of "Cross-Sell Products" (dash)

File: Model/StringFormatter.php [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    }
+
+    /**
+     * This unique ID can only contain alphanumeric characters (letters and numbers
+     * only) and also the asterisk, hyphen, and underscore characters. If your
+     * product IDs contain invalid characters, simply replace them with an alternate
+     * character like an underscore. This will only be used in the feed and not for
+     * any customer facing purpose.
+     *
+     * @static
```
Comment (brobie) [inline Model/StringFormatter.php]: Why is this declared as static, but some of the other functions like jsonEncode are not?

File: Model/XMLWriter.php [hunk 4]
```diff
@@ -31,14 +29,19 @@ public function writeElement($name, $content = null, $cdata = false)
...
+    /**
+     * @param null $content
+     * @param bool $cdata
+     *
+     * @return bool|void
+     */
     public function writeRaw($content = null, $cdata = false)
     {
-	    $content = trim($content);
+        $content = trim($content);
```
Comment (brobie) [inline Model/XMLWriter.php]: should you cast $content to a string here?  If not, why above?

File: Plugin/ProductList/Crosssell.php [hunk 5]
```diff
@@ -1,4 +1,10 @@
 <?php
+declare(strict_types=1);
+
+namespace Bazaarvoice\Connector\Plugin\ProductList;
+
+use Bazaarvoice\Connector\Model\Source\ProductList;
+
 /**
  * StoreFront Bazaarvoice Extension for Magento
```
Comment (brobie) [inline Plugin/ProductList/Crosssell.php]: Should remove all references to "StoreFront"

File: Plugin/ProductList/Item.php [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    public function __construct(
+        ConfigProviderInterface $configProvider,
+        StringFormatterInterface $stringFormatter
+    ) {
+        $this->configProvider = $configProvider;
+        $this->stringFormatter = $stringFormatter;
+    }
+
+    // @codingStandardsIgnoreStart
+    public function beforeGetProductPrice(
```
Comment (brobie) [inline Plugin/ProductList/Item.php]: If you are implementing a "before" and "after", why not use an "around"?

File: Plugin/ProductList/Related.php [hunk 7]
```diff
@@ -1,4 +1,10 @@
 <?php
+declare(strict_types=1);
+
+namespace Bazaarvoice\Connector\Plugin\ProductList;
+
+use Bazaarvoice\Connector\Model\Source\ProductList;
+
 /**
  * StoreFront Bazaarvoice Extension for Magento
```
Comment (brobie) [inline Plugin/ProductList/Related.php]: Remove StoreFront comments

File: Plugin/ProductList/Widget.php [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+     */
+    protected $type = ProductList::WIDGET;
+
+    // @codingStandardsIgnoreStart
+
+    /**
+     * @param $subject
+     * @param $product
+     */
+    public function beforeGetProductPriceHtml(
```
Comment (brobie) [inline Plugin/ProductList/Widget.php]: Use around plugin?

File: Setup/InstallData.php [hunk 9]
```diff
@@ -1,34 +1,25 @@
...
-
+use Magento\Sales\Setup\SalesSetup;
+use Magento\Sales\Setup\SalesSetupFactory;
 
+/**
+ * Class InstallData
+ *
+ * @package Bazaarvoice\Connector\Setup
+ */
 class InstallData implements Setup\InstallDataInterface
```
Comment (brobie) [inline Setup/InstallData.php]: I wonder how we support declarative schema?  Probably not possible until the older versions of Magento are not supported anymore.

File: Setup/InstallData.php [hunk 10]
```diff
@@ -76,29 +67,29 @@ public function install(Setup\ModuleDataSetupInterface $setup, Setup\ModuleConte
...
+                'visible'                 => true,
+                'required'                => false,
+                'user_defined'            => false,
+                'default'                 => Boolean::VALUE_NO,
+                'apply_to'                => '',
+                'visible_on_front'        => false,
+                'is_used_in_grid'         => true,
+                'is_visible_in_grid'      => false,
+                'is_filterable_in_grid'   => false,
+                'used_in_product_listing' => true,
```
Comment (brobie) [inline Setup/InstallData.php]: No need for trailing comma

File: Setup/UpgradeData.php [hunk 11]
```diff
@@ -46,58 +34,58 @@ public function __construct(
...
+                    'visible'                 => true,
+                    'required'                => false,
+                    'user_defined'            => false,
+                    'default'                 => '',
+                    'apply_to'                => '',
+                    'visible_on_front'        => false,
+                    'is_used_in_grid'         => true,
+                    'is_visible_in_grid'      => false,
+                    'is_filterable_in_grid'   => false,
+                    'used_in_product_listing' => true,
```
Comment (brobie) [inline Setup/UpgradeData.php]: trailing comma

File: build.xml [hunk 12]
```diff
@@ -1,129 +1,109 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (brobie) [inline build.xml]: Is this Ant script even needed anymore?

File: Api/StringFormatterInterface.php [hunk 13]
```diff
@@ -0,0 +1,92 @@
+<?php
+
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Api/StringFormatterInterface.php]: Yes. Added.

File: Api/StringFormatterInterface.php [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+/**
+ * Interface StringFormatterInterface
+ *
+ * @package Bazaarvoice\Connector\Api
+ */
+interface StringFormatterInterface
+{
+    /**
+     * This unique ID can only contain alphanumeric characters (letters and numbers
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Api/StringFormatterInterface.php]: Agreed. Fixed.

File: Console/Command/Export.php [hunk 15]
```diff
@@ -1,37 +1,33 @@
...
- * @author    Dennis Rogers <dennis@storefrontconsulting.com>
- * @copyright 2016 StoreFront Consulting, Inc
- * @license   http://www.storefrontconsulting.com/media/downloads/ExtensionLicense.pdf StoreFront Consulting Commercial License
- * @link      http://www.StoreFrontConsulting.com/bazaarvoice-extension/
- */
+declare(strict_types=1);
 
 namespace Bazaarvoice\Connector\Console\Command;
 
+use Exception;
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Console/Command/Export.php]: All FQNs should be simplified according to Magento's PHPCS rules. So this is flagged if it is not imported.

File: Console/Command/Export.php [hunk 16]
```diff
@@ -41,16 +37,23 @@ protected function configure()
...
         try {
             $this->_exporter->exportReviews();
-        } Catch (\Exception $e) {
-            echo $e->getMessage() . "\n" . $e->getTraceAsString();
+        } catch (Exception $e) {
+            echo $e->getMessage()."\n".$e->getTraceAsString();
         }
-        echo "\n" . 'Memory usage: ' . memory_get_usage() . "\n";
+        echo "\n".'Memory usage: '.memory_get_usage()."\n";
+        // @codingStandardsIgnoreEnd
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Console/Command/Export.php]: Agreed. Fixed.

File: Console/Command/Purchase.php [hunk 17]
```diff
@@ -43,12 +35,18 @@ protected function configure()
...
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     *
+     * @return int|void|null
+     */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        // @codingStandardsIgnoreEnd
-        echo "\n" . 'Memory usage: ' . memory_get_usage() . "\n";
+        echo "\n".'Memory usage: '.memory_get_usage()."\n";
         $this->_purchaseFeed->generateFeed();
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Console/Command/Purchase.php]: I do not think so, generateFeed does not throw an exception.

File: Console/Command/PurchaseTest.php [hunk 18]
```diff
@@ -43,12 +35,18 @@ protected function configure()
...
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface   $input
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     *
+     * @return int|void|null
+     */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
+        echo "\n".'Memory usage: '.memory_get_usage()."\n";
+        $this->_purchaseFeed->setTest(true)->generateFeed();
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Console/Command/PurchaseTest.php]: I do not think so, generateFeed does not throw an exception.

File: Logger/Logger.php [hunk 19]
```diff
@@ -1,87 +1,90 @@
...
-            $message = print_r($message, 1);
+        if (is_array($message)) {
+            // phpcs:ignore
+            $message = print_r($message, $return = true);
+        }
 
-        if (
-        	php_sapi_name() == "cli"
-	        || $this->admin
+        if (php_sapi_name() == "cli"
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Logger/Logger.php]: No. Fixed.

File: Model/Feed/ProductFeed.php [hunk 20]
```diff
@@ -1,147 +1,151 @@
...
     public function exportFeedForGlobal()
     {
         /** Using admin store for now */
-        /** @var \Magento\Store\Model\StoreManagerInterface $storeManager */
-        $storeManager = $this->_objectManager->get('Magento\Store\Model\StoreManagerInterface');
         /** @var Store $store */
-        $store = $storeManager->getStore(0);
+        $store = $this->storeManager->getStore(0);
 
         /** Create varien io object and write local feed file */
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Model/Feed/ProductFeed.php]: No. Fixed.

File: Model/Index.php [hunk 21]
```diff
@@ -49,6 +70,7 @@ public function getIdentities()
      * @param null $scope
      *
      * @return $this|mixed
+     * @throws \Magento\Framework\Exception\LocalizedException
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Model/Index.php]: I do not think so, at least according to the code styles from PSR1/PSR2 that I'm using. And I am seeing that Magento\Framework does not include a new line.

File: Model/Notification.php [hunk 22]
```diff
@@ -53,30 +46,30 @@ public function isDisplayed()
...
-        $finished = new \DateTime($last->getFinishedAt());
-
-        if (
-            $now->diff($executed)->format('%a') > 10 ||
-            $now->diff($finished)->format('%a') > 10
-        )
-            return true;
-
-        return false;
+         * $schedule = $this->_scheduleFactory->create()->getCollection();
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Model/Notification.php]: Agreed. Removed.

File: Controller/Adminhtml/Bvfeed/Runproduct.php [hunk 23]
```diff
@@ -1,45 +1,43 @@
...
-        $this->_productFeed = $productFeed;
     }
 
+    /**
+     * @return \Magento\Framework\App\ResponseInterface|\Magento\Framework\Controller\ResultInterface|void
+     */
     public function execute()
     {
+        // phpcs:ignore
         echo '<pre>';
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Controller/Adminhtml/Bvfeed/Runproduct.php]: I have added a todo to investigate this.

File: Model/SeoContent.php [hunk 24]
```diff
@@ -0,0 +1,200 @@
...
+        $this->currentProductProvider = $currentProductProvider;
+        $this->stringFormatter = $stringFormatter;
+        $this->requestInterface = $requestInterface;
+        $this->urlInterface = $urlInterface;
+    }
+
+    /**
+     * @return string
+     */
+    public function getQaSeoContent()
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Model/SeoContent.php]: Agreed. Fixed.

File: Model/SeoContent.php [hunk 25]
```diff
@@ -0,0 +1,200 @@
...
+            $seoContent .= $this->getDebugSeoParams();
+            return $seoContent;
+        }
+        return '';
+    }
+
+    /**
+     * @return false|string
+     */
+    public function getRrSeoContent()
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Model/SeoContent.php]: Agreed. Fixed.

File: Model/SeoContent.php [hunk 26]
```diff
@@ -0,0 +1,200 @@
...
+
+        return $this->seoSdk;
+    }
+
+    /**
+     * @return string
+     */
+    private function getDebugSeoParams()
+    {
+        if ($this->configProvider->getEnvironment() == 'staging') {
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Model/SeoContent.php]: Agreed. Fixed.

File: Model/SeoContent.php [hunk 27]
```diff
@@ -0,0 +1,200 @@
...
+            'seo_sdk_enabled' => true,
+            'bv_root_folder'  => $deploymentZoneId,
+            /** replace with your display code (BV provided) */
+            'subject_id'      => $this->stringFormatter->getFormattedProductSku($product),
+            /** replace with product id */
+            'cloud_key'       => $this->configProvider->getCloudSeoKey(),
+            /** BV provided value */
+            'base_url'        => $baseUrl,
+            'page_url'        => $productUrl,
+            'staging'         => ($this->configProvider->getEnvironment() == 'staging' ? true : false),
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Model/SeoContent.php]: Agreed. Fixed.

File: Model/Source/Category.php [hunk 28]
```diff
@@ -1,109 +1,113 @@
...
+                    $names[$category->getId()] = $category->getName();
+                    $name = implode('/', $names);
+                    $this->categories[] = [
+                        'value' => $category->getId(),
+                        'label' => $name,
+                    ];
+                }
+                //phpcs:ignore
+            } catch (LocalizedException $e) {
+                //no-op
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Model/Source/Category.php]: Agreed. Fixed.

File: Model/Source/ProductAttribute.php [hunk 29]
```diff
@@ -58,39 +68,44 @@ public function toOptionArray()
...
         }
 
-        $attributeOptions = array(array(
-            'label' => __('-- Please Select --'),
-            'value' => ''
-        ));
+        if (!$isMultiselect) {
+            $attributeOptions = [
+                [
+                    'label' => __('-- Please Select --'),
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Model/Source/ProductAttribute.php]: Fixed discrepancy. Only constant in core that I see is in Magento\Sales.

File: Model/Source/ProductList.php (hunk 1 above)
Comment (jantzenw) [reply Model/Source/ProductList.php]: Fixed.

File: Model/Source/ProductList.php (hunk 2 above)
Comment (jantzenw) [reply Model/Source/ProductList.php]: Fixed.

File: Model/StringFormatter.php (hunk 3 above)
Comment (jantzenw) [reply Model/StringFormatter.php]: Removed @static. Not sure where that came from.

File: Model/XMLWriter.php (hunk 4 above)
Comment (jantzenw) [reply Model/XMLWriter.php]: Yes. Fixed.

File: Plugin/ProductList/Crosssell.php (hunk 5 above)
Comment (jantzenw) [reply Plugin/ProductList/Crosssell.php]: Fixed in other PR.

File: Plugin/ProductList/Related.php (hunk 7 above)
Comment (jantzenw) [reply Plugin/ProductList/Related.php]: Fixed in other PR.

File: Plugin/ProductList/Widget.php (hunk 8 above)
Comment (jantzenw) [reply Plugin/ProductList/Widget.php]: I do not see a reason why not. This code was originally written by StoreFront. I have added a todo to test combining these into an around plugin.

File: Setup/InstallData.php (hunk 9 above)
Comment (jantzenw) [reply Setup/InstallData.php]: Not sure. Agreed.

File: Setup/InstallData.php (hunk 10 above)
Comment (jantzenw) [reply Setup/InstallData.php]: Fixed.

File: Setup/UpgradeData.php (hunk 11 above)
Comment (jantzenw) [reply Setup/UpgradeData.php]: Fixed.

File: build.xml (hunk 12 above)
Comment (jantzenw) [reply build.xml]: I don't think so. Removed.

File: Plugin/ProductList/Item.php (hunk 6 above)
Comment (jantzenw) [reply Plugin/ProductList/Item.php]: I do not see a reason why not. This code was originally written by StoreFront. I have added a todo to test combining these into an around plugin.

File: Api/StringFormatterInterface.php (hunk 13 above)
[the remark this answers is not in the document]
Comment (jantzenw) [reply Api/StringFormatterInterface.php]: Yes. Added.

File: Api/StringFormatterInterface.php (hunk 14 above)
[the remark this answers is not in the document]
Comment (jantzenw) [reply Api/StringFormatterInterface.php]: Agreed. Fixed.

File: Console/Command/Export.php (hunk 15 above)
[the remark this answers is not in the document]
Comment (jantzenw) [reply Console/Command/Export.php]: All FQNs should be simplified according to Magento's PHPCS rules. So this is flagged if it is not imported.

File: Console/Command/Export.php (hunk 16 above)
[the remark this answers is not in the document]
Comment (jantzenw) [reply Console/Command/Export.php]: Agreed. Fixed.

File: Console/Command/Purchase.php (hunk 17 above)
[the remark this answers is not in the document]
Comment (jantzenw) [reply Console/Command/Purchase.php]: I do not think so, generateFeed does not throw an exception.

File: Plugin/ProductList/Widget.php (hunk 8 above)
Comment (jantzenw) [reply Plugin/ProductList/Widget.php]: I do not think so, generateFeed does not throw an exception.

File: Setup/InstallData.php (hunk 10 above)
Comment (jantzenw) [reply Setup/InstallData.php]: No. Fixed.

File: Setup/UpgradeData.php (hunk 11 above)
Comment (jantzenw) [reply Setup/UpgradeData.php]: Fixed.

File: build.xml (hunk 12 above)
Comment (jantzenw) [reply build.xml]: I do not think so, at least according to the code styles from PSR1/PSR2 that I'm using. And I am seeing that Magento\Framework does not include a new line.

**Pull request merged**